source/agora/node/Ledger.d

```diff
@@ -166,12 +184,9 @@ public class Ledger
 
     ***************************************************************************/
 
-    private bool isValidBlock (const ref Block block) pure nothrow @safe @nogc
+    private bool isValidBlock (const ref Block block) nothrow @safe
     {
-        const expected_height = this.last_block !is null
-            ? (this.last_block.header.height + 1)
-            : 0;
-
+        const expected_height = this.block_height + 1;
```

Comment:
You can remove the constant if it's only used in one place. Just write:

```D
return block.header.height == this.block_height + 1;
```

Reply:
Applied the suggestion